### DISCUSSION TITLE
Week 3 – On the way to the first Prototype

### DISCUSSION THREAD

**Discussion by Mohmad-Naser-alnakeshbandi** [The-Escape-Box/TheEscapeBox#21] [General]
This week we focused on implementing more features and implementing the missing core mechanics of the game

# Weekly Progress

This aim of this week is, developing the enemy AI and increase the environment, and guess what we got it 🚀

## Environment:

![image](https://github.com/The-Escape-Box/TheEscapeBox/assets/76824323/29071e0d-7039-456c-85a3-35fec70c2e4f)

![image](https://github.com/The-Escape-Box/TheEscapeBox/assets/76824323/3721e9bd-a56f-4120-8e6f-81676b6b2502)

## But how should the enmies chase the player and kill him?

Well, we use what it called "AI Navigation" in unity. You can read more about it here: https://docs.unity3d.com/Packages/com.unity.ai.navigation@2.0/manual/

But to sum it up:

- you just need to define "Walkable Areas" 

![image](https://github.com/The-Escape-Box/TheEscapeBox/assets/76824323/bef0c34e-7a07-488c-aaf9-d3a747c41f69)

- And "NavMesh Agent" that chase your player:

![image](https://github.com/The-Escape-Box/TheEscapeBox/assets/76824323/d8a93e4d-3e38-4891-a9e8-84ea591af850)

--- 
Time
Mohammad: 20h
Lukas: 18h

---
Please feel free to share your thoughts on the project in this and all subsequent posts. We will consistently update you on the steps we have taken and our plans for the future. Our aim is to fulfill both our own and your expectations.

**Comment by tonoizer** [comment]
Great strides in Week 3, indeed! The addition of enemy AI and expanding the environment is commendable. However, have you encountered any challenges or roadblocks during the implementation process? How do you plan to balance the difficulty of enemy encounters to ensure an engaging gameplay experience for players of all skill levels?

Team: AetherArena

**Comment by TimWortmann** [comment]
Hey team!
I'm absolutely blown away by the progress you've made this week. Implementing more features and fleshing out the missing core mechanics of the game is no small feat, but you've tackled it head-on with determination and skill. The environment you've created looks stunning! It's immersive and captivating, setting the perfect stage for the thrilling adventures that await players. And the addition of enemy AI adds a whole new layer of excitement and challenge to the game. Using AI Navigation in Unity to make enemies chase and confront the player is a brilliant move. It adds a sense of realism and urgency to the gameplay, making every encounter feel dynamic and engaging. I'm genuinely impressed by the level of detail and thoughtfulness you're putting into this project.

Team
Cooking Souls
